### ISSUE TITLE
[Question]: How to create a multiDocagent using function call with bedrock llms

### ISSUE THREAD

**Issue by noviljohnson** [run-llama/llama_index#13381]
### Question Validation

- [X] I have searched both the documentation and discord for an answer.

### Question

I want to create multi document agents using function calling as shown in here [structuredPlannerAgent](https://docs.llamaindex.ai/en/stable/examples/agent/structured_planner/#high-level-api), but with bedrock llms ("anthropic.claude-v2",
"anthropic.claude-3-sonnet-20240229-v1:0")

but i am getting an error that these models don't support function calling. 
i checked the `Settings.llm.metadata.is_function_calling_model` is `False`

1. is it possible to make the `Settings.llm.metadata.is_function_calling_model`  :  `True` and use the above-mentioned models for function calling
2. Or suggest some Bedrock llms that can be used for Function Calling to use in [structuredPlannerAgent](https://docs.llamaindex.ai/en/stable/examples/agent/structured_planner/#high-level-api)
3. Or is there any other way to use above mentioned models and create multi document agents.

I tried this  [Multidocagents](https://docs.llamaindex.ai/en/stable/examples/agent/multi_document_agents/) as well, but with `ReActAgent` instead of `OpenAiAgent`.
but the output is not as expected. I can't use Openai api and i have to go with bedrock llms.

basically, i am trying to create the following workflow.
![agents workflow](https://github.com/run-llama/llama_index/assets/41065220/30a41c84-b0fe-47aa-9ec2-7b9c20d32760)

**Comment by logan-markewich** [comment]
You need to use the `ReactAgentWorker`, since the bedrock LLM class doesn't implement the `FunctionCallingLLM` class

**Comment by logan-markewich** [comment]
```python
from llama_index.core.agent import (
    StructuredPlannerAgent,
    ReActAgentWorker,
)

# create the react agent worker for reasoning
worker = ReActAgentWorker.from_tools(
    [lyft_tool, uber_tool], verbose=True
)

# wrap the worker in the top-level planner
agent = StructuredPlannerAgent(
    worker, tools=[lyft_tool, uber_tool], verbose=True
)
```

**Comment by noviljohnson** [comment]
Hi @logan-markewich, 
thank you

it worked with `ReactAgentWorker`.

but i am getting response sometimes in English and sometimes in different language (maybe French/ Latin 😂) 
could you tell me why this is happening. Or should i change anything.



thankx

**Comment by noviljohnson** [comment]
hi,
after experimenting for some time, now i am getting all responses in English.
But it is taking 1.5 to 2 min to return the response.  Is this normal response time?.
Is there a way to reduce the response time.


Thank you.

**Comment by logan-markewich** [comment]
The response time depends on the LLM being used, how many LLM calls are being made, and how much text the LLM is writing.

Typically there is
1 LLM call to either call a tool or respond
1 tool call (which could involve 1+ LLM calls)
1 LLM call to either respond using the tool response or call another tool
and so on

**Comment by noviljohnson** [comment]
Thank you @logan-markewich
